### PR TITLE
fix: [CDS-83592]: add sidenav footer unit tests

### PR DESCRIPTION
**Pull request by MondalSayantan**
### Summary

Tested
- should render side nav footer
- should open profile popover on hover
- should open profile overview page
- should open resource center drawer
- should open profile overview page on clicking footer link
- should signout

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by MondalSayantan** [comment]
retrigger eslint

File: src/modules/10-common/navigation/SideNavV2/SideNavFooter/__tests__/SideNavFooter.test.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+import React from 'react'
+import { render, waitFor, screen, RenderResult } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { TestWrapper, findDrawerContainer, findPopoverContainer } from '@common/utils/testUtils'
+import SideNavFooter from '../SideNavFooter'
+
+const mockHistoryPush = jest.fn()
+// eslint-disable-next-line jest-no-mock
+jest.mock('react-router-dom', () => ({
```
Comment (Sagar-Rajput7) [inline src/modules/10-common/navigation/SideNavV2/SideNavFooter/__tests__/SideNavFooter.test.tsx:16]: We should not mock react-router-dom.

Comment (Sagar-Rajput7) [reply src/modules/10-common/navigation/SideNavV2/SideNavFooter/__tests__/SideNavFooter.test.tsx:16]: https://github.com/harness/harness-core-ui/blob/deb92e66bab0523fac0389b30c21870950380527/.eslintrc.yml#L51

File: src/modules/10-common/navigation/SideNavV2/SideNavFooter/__tests__/SideNavFooter.test.tsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    const user = userEvent.setup()
+    await user.hover(queryByText('Dev Name') as HTMLElement)
+    await waitFor(() => {
+      const popover = findPopoverContainer() as HTMLElement
+      expect(popover).toBeInTheDocument()
+    })
+    await user.click(screen.getByText('signOut'))
+    expect(mockHistoryPush).toHaveBeenCalledWith({
+      pathname: '/redirect',
+      search: '?returnUrl=%2F%23%2Flogin%3Faction%3Dsignout'
```
Comment (Sagar-Rajput7) [inline src/modules/10-common/navigation/SideNavV2/SideNavFooter/__tests__/SideNavFooter.test.tsx:132]: NIT: can also use decode value for better readability

File: src/modules/10-common/navigation/SideNavV2/SideNavFooter/__tests__/SideNavFooter.test.tsx (hunk 1 above)
Comment (MondalSayantan) [reply src/modules/10-common/navigation/SideNavV2/SideNavFooter/__tests__/SideNavFooter.test.tsx:16]: I need to mock `react-router-dom ` to check if history.push has been called with the correct values. This approach has been used in multiple places eg: https://github.com/harness/harness-core-ui/blob/develop/src/modules/70-pipeline/utils/__tests__/withInputSetsOnCreateUpdateSuccess.test.tsx

**Review by avinashmadhwani02 — APPROVED** [review]

File: src/modules/10-common/navigation/SideNavV2/SideNavFooter/__tests__/SideNavFooter.test.tsx (hunk 1 above)
Comment (avinashmadhwani02) [reply src/modules/10-common/navigation/SideNavV2/SideNavFooter/__tests__/SideNavFooter.test.tsx:16]: i think it's fine to mock it. @Sagar-Rajput7 Is there any alternative method to achieve this?

**Review by Sagar-Rajput7 — APPROVED** [review]

File: src/modules/10-common/navigation/SideNavV2/SideNavFooter/__tests__/SideNavFooter.test.tsx (hunk 1 above)
Comment (chetanvekariya-harness) [reply src/modules/10-common/navigation/SideNavV2/SideNavFooter/__tests__/SideNavFooter.test.tsx:16]: We should check for final url for assertion. We should not care how history.push was called in this case.

**Pull request merged**